runtime/testy/runtime.py

```diff
@@ -103,17 +118,100 @@ def startStateManager(badThingsQueue, stateQueue, runtimePipe):
   except Exception:
     badThingsQueue.put(BadThing(sys.exc_info(), None))
 
-def processFactory(badThingsQueue, stateQueue):
-  def spawnProcessHelper(processName, helper):
+def processFactory(badThingsQueue, stateQueue, stdoutRedirect = None):
+  def spawnProcessHelper(processName, helper, *args):
     pipeToChild, pipeFromChild = multiprocessing.Pipe()
     if processName != PROCESS_NAMES.STATE_MANAGER:
       stateQueue.put([SM_COMMANDS.ADD, [processName, pipeToChild]], block=True)
       pipeFromChild.recv()
-    newProcess = multiprocessing.Process(target=helper, name=processName.value, args=(badThingsQueue, stateQueue, pipeFromChild))
+    newProcess = multiprocessing.Process(target=helper, name=processName.value, args=[badThingsQueue, stateQueue, pipeFromChild] + list(args))
     allProcesses[processName] = newProcess
     newProcess.daemon = True
     newProcess.start()
   return spawnProcessHelper
 
+def runtimeTest():
+  # Normally dangerous. Allowed here because we put testing code there.
+  import studentCode
+
+  testNameRegex = re.compile(".*_setup")
+  testNames = [testName[:-len("_setup")] for testName in dir(studentCode) if testNameRegex.match(testName)]
+
+  failCount = 0
+  failedTests = []
+
+  for testName in testNames:
+    testFileName = "%s_output" % (testName,)
+    with open(testFileName, "w") as testOutput:
+      class Unbuffered:
```

Comment:
If you're on Python 3, have you tried the following: https://docs.python.org/3/library/io.html#raw-i-o

Perhaps this solution to the buffering problem obsoletes the mailing list solution from 2003.